### PR TITLE
Introduce find_mutation and mutation_stats class

### PR DESCRIPTION
**Pull request by stevenhwu**
Replace #104 
find_mutation: used in call.cc, call peeling operation, and pass workspace to mutation_stats. Some unit tests.

mutation_stats: calculate and store all stats. Store stats to vcf. Some unit tests
TODO(SW): need an interface between mutation_stats and outfile(various format)
TODO(SW): Later on will have switches allow users only output a subsets of stats. Still working on moving some counts related stats from call.cc to here.

Refactoing peeling.cc. Issue #80. Fixed merge issue on #83 

test/src/fixture/\* - Collection of fixtures can be used in multiple boost_tests.

### CODE REVIEW

**Comment by reedacartwright** [comment]
Jenkins, ok to test

**Comment by stevenhwu** [comment]
Why this error appear now but not before? Were some headers got included implicitly?

Address Jenkins' error

``` c++
In file included from /usr/local/jenkins/jobs/DeNovoGear-PR/workspace/src/include/dng/hts/bcf.h:23:
/usr/local/jenkins/jobs/DeNovoGear-PR/workspace/src/include/dng/hts/hts.h:56:95: error: use of undeclared identifier 'free'
        std::unique_ptr<char[], void(*)(void *)> s{hts_format_description(&handle()->format), free};
                                                                                              ^
```

Fixed with `#include <cstdlib>` in `dng/hts/hts.h`

**Comment by stevenhwu** [comment]
Just want to double check this, does Jenkin ran `make test` or see if code in the test folder compiles without error?
I can find `[build] $ /usr/bin/make` but not `make other_targets` in Jenkins' log.

**Comment by reedacartwright** [comment]
Currently, our Jenkins instance runs on a FreeBSD 10 machine with Clang 3.4 and libc++.  Something that works on Linux development boxes can fail on the Jenkins machine.

Additionally, our Jenkins instance just checks for successful compiling and not successful tests. As soon as #35 is solved, we will do more.

**Comment by stevenhwu** [comment]
but `src/include/dng/hts/hts.h` was last modified on June 2015 and pass all previous tests/compile. Are there any ways to capture more issues like this?

**Comment by reedacartwright** [comment]
The problem was masked earlier due to `cstdlib` being included before `hts.h`.  But the new `mutation_stats.cc` file exposed it.

As far as catching these, we when doing unit testing on headers, we should make the header the first one included.

File: src/include/dng/mutation_stats.h [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    void RecordBasicStats(hts::bcf::Variant &record);
+
+    void RecordGenotypeStats(hts::bcf::Variant &record);
+
+    void RecordSingleMutationStats(hts::bcf::Variant &record);
+
+
+    enum class OutputLevel {
+        Basic, Complete, Singcle
+    };
```
Comment (reedacartwright) [inline src/include/dng/mutation_stats.h:84]: Misspelling

**Pull request closed without merging**